presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/EliminateCrossJoins.java

```diff
@@ -145,14 +146,14 @@ public Rewriter(PlanNodeIdAllocator idAllocator, JoinGraph graph, List<Integer>
         {
             this.idAllocator = requireNonNull(idAllocator, "idAllocator is null");
             this.graph = requireNonNull(graph, "graph is null");
-            this.joinOrder = requireNonNull(joinOrder, "joinOrder is null");
+            this.joinOrder = ImmutableList.copyOf(requireNonNull(joinOrder, "joinOrder is null"));
```

Comment:
It does, but the exc msg would be less informative. I think I follow established convention here.